osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs

```diff
@@ -135,6 +136,87 @@ public async Task TestImportThenImport()
             }
         }
 
+        [Test]
+        public async Task TestExportBeatMapWithInvalidCharactersPath()
+        {
+            using (HeadlessGameHost host = new CleanRunHeadlessGameHost(nameof(ImportBeatmapTest)))
+            {
+                try
+                {
+                    var osu = LoadOsuIntoHost(host);
+
+                    var temp = TestResources.GetTestBeatmapForImport();
+                    //Folder to export to
+                    string extractedFolder = $"{temp}_extracted";
+                    Directory.CreateDirectory(extractedFolder);
+
+                    //Full path to the exported .osz file (the last part of the path gets added on Line 188 after the beatmap gets loaded)
+                    string exportFullPath = extractedFolder;
+
+                    try
+                    {
+                        //Lists for storing the content of .osz archive before and after being exported
+                        List<string> firstOszFiles = new List<string>();
+                        List<string> secondOszFiles = new List<string>();
+
+                        //Before Export
+                        using (var firstZip = ZipFile.Open(temp, ZipArchiveMode.Read))
+                        {
+                            foreach (var entry in firstZip.Entries)
+                                firstOszFiles.Add(entry.FullName);
+                        }
+
+                        var firstImport = await LoadOszIntoOsu(osu, temp);
+
+                        //Clone firstImport data to modify the path to match the path construction created in osu.Game/Beatmaps/BeatmapManager.cs function Save() Line 270
+                        var cloneFirstImport = firstImport;
+                        var metadata = cloneFirstImport.Metadata;
+
+                        foreach (var beatmapInfo in cloneFirstImport.Beatmaps)
+                        {
+                            //Construct path as in osu.Game/Beatmaps/BeatmapManager.cs function Save() Line 270
+                            //The extra \\ are added in case the beatmap metadata doesn't contain invalid characters, it ensures that the path has invalid characters
+                            //Before applying the fix on line 183
+                            beatmapInfo.Path = $"{metadata.Artist} - {metadata.Title} ({metadata.Author}) [\\{beatmapInfo.Version}].osu";
+
+                            //Remove invalid characters as done in the same Save() function by calling the protected static function GetValidFilename()
+                            foreach (char c in Path.GetInvalidFileNameChars())
+                                beatmapInfo.Path = beatmapInfo.Path.Replace(c, '_');
+                        }
+
+                        // Exporting the new cloneFirstImport
+                        exportFullPath += $"\\{cloneFirstImport}.osz";
+                        exportTestBeatmap(osu, cloneFirstImport, exportFullPath);
```

Comment:
this is unacceptable. this is explicitly bypassing *all of the code this is supposed to be testing* by reimplementing both the problem scenario and the fix in the test. therefore this test, in fact, tests *absolutely nothing of use*. i'm 99% certain i could revert your _actual_ fix and still have this test pass.

i'm sorry but i'm not willing to review this pull any more than i already have, either. a change as simple as this should never have undergone so many rounds of grueling review.